[main] Update dependencies from dotnet/runtime dotnet/arcade dnceng/internal/dotnet-optimization dotnet/xharness dotnet/hotreload-utils

ViktorHofer opened the pull request:
Another attempt of #54060

dotnet-maestro[bot] committed:
Update dependencies from https://github.com/dotnet/xharness build 20210610.3

Microsoft.DotNet.XHarness.CLI , Microsoft.DotNet.XHarness.TestRunners.Xunit
 From Version 1.0.0-prerelease.21307.1 -> To Version 1.0.0-prerelease.21310.3

dotnet-maestro[bot] committed:
Update dependencies from https://github.com/dotnet/hotreload-utils build 20210610.2

Microsoft.DotNet.HotReload.Utils.Generator.BuildTool
 From Version 1.0.1-alpha.0.21307.1 -> To Version 1.0.1-alpha.0.21310.2

dotnet-maestro[bot] committed:
Update dependencies from https://dev.azure.com/dnceng/internal/_git/dotnet-optimization build 20210610.4

optimization.linux-x64.MIBC.Runtime , optimization.windows_nt-x64.MIBC.Runtime , optimization.windows_nt-x86.MIBC.Runtime , optimization.PGO.CoreCLR
 From Version 1.0.0-prerelease.21308.4 -> To Version 1.0.0-prerelease.21310.4

dotnet-maestro[bot] committed:
Update dependencies from https://github.com/dotnet/arcade build 20210611.3

Microsoft.DotNet.XUnitExtensions , Microsoft.DotNet.VersionTools.Tasks , Microsoft.DotNet.Build.Tasks.TargetFramework.Sdk , Microsoft.DotNet.Build.Tasks.Packaging , Microsoft.DotNet.Build.Tasks.Installers , Microsoft.DotNet.Build.Tasks.Feed , Microsoft.DotNet.Build.Tasks.Archives , Microsoft.DotNet.Arcade.Sdk , Microsoft.DotNet.ApiCompat , Microsoft.DotNet.CodeAnalysis , Microsoft.DotNet.XUnitConsoleRunner , Microsoft.DotNet.GenFacades , Microsoft.DotNet.GenAPI , Microsoft.DotNet.RemoteExecutor , Microsoft.DotNet.PackageTesting , Microsoft.DotNet.Helix.Sdk , Microsoft.DotNet.SharedFramework.Sdk
 From Version 6.0.0-beta.21304.1 -> To Version 6.0.0-beta.21311.3

dotnet-maestro[bot] committed:
Update dependencies from https://github.com/mono/linker build 20210610.3

Microsoft.NET.ILLink.Tasks
 From Version 6.0.100-preview.6.21308.1 -> To Version 6.0.100-preview.6.21310.3

dotnet-maestro[bot] committed:
Update dependencies from https://github.com/dotnet/hotreload-utils build 20210611.1

Microsoft.DotNet.HotReload.Utils.Generator.BuildTool
 From Version 1.0.1-alpha.0.21307.1 -> To Version 1.0.1-alpha.0.21311.1

dotnet-maestro[bot] committed:
Update dependencies from https://dev.azure.com/dnceng/internal/_git/dotnet-optimization build 20210612.4

optimization.linux-x64.MIBC.Runtime , optimization.windows_nt-x64.MIBC.Runtime , optimization.windows_nt-x86.MIBC.Runtime , optimization.PGO.CoreCLR
 From Version 1.0.0-prerelease.21308.4 -> To Version 1.0.0-prerelease.21312.4

dotnet-maestro[bot] committed:
Update dependencies from https://github.com/dotnet/runtime build 20210614.1

Microsoft.NETCore.DotNetHost , Microsoft.NETCore.DotNetHostPolicy , Microsoft.NETCore.ILAsm , runtime.native.System.IO.Ports , Microsoft.NET.Sdk.IL , System.Runtime.CompilerServices.Unsafe , System.Text.Json
 From Version 6.0.0-preview.6.21307.1 -> To Version 6.0.0-preview.6.21314.1

dotnet-maestro[bot] committed:
Update dependencies from https://dev.azure.com/dnceng/internal/_git/dotnet-optimization build 20210613.4

optimization.linux-x64.MIBC.Runtime , optimization.windows_nt-x64.MIBC.Runtime , optimization.windows_nt-x86.MIBC.Runtime , optimization.PGO.CoreCLR
 From Version 1.0.0-prerelease.21308.4 -> To Version 1.0.0-prerelease.21313.4

dotnet-maestro[bot] committed:
Update dependencies from https://github.com/dotnet/xharness build 20210614.1

Microsoft.DotNet.XHarness.CLI , Microsoft.DotNet.XHarness.TestRunners.Xunit
 From Version 1.0.0-prerelease.21307.1 -> To Version 1.0.0-prerelease.21314.1

ViktorHofer committed:
Update Versions.props

ViktorHofer committed:
Update Versions.props

hoyosjs commented on eng/Version.Details.xml:
```diff
@@ -146,73 +146,73 @@
+      <Sha>af5c238556e204583b129cc8f5c7338f84dc2c40</Sha>
     </Dependency>
-    <Dependency Name="Microsoft.NET.ILLink.Tasks" Version="6.0.100-preview.6.21308.1">
+    <Dependency Name="Microsoft.NET.ILLink.Tasks" Version="6.0.100-preview.6.21310.3">
```
```suggestion
    <Dependency Name="Microsoft.NET.ILLink.Tasks" Version="6.0.100-preview.6.21308.1">
```

hoyosjs commented on eng/Version.Details.xml:
```diff
@@ -146,73 +146,73 @@
+    <Dependency Name="Microsoft.NET.ILLink.Tasks" Version="6.0.100-preview.6.21310.3">
       <Uri>https://github.com/mono/linker</Uri>
-      <Sha>802aa459394baa2451ffacc02219ba54ef77ca44</Sha>
+      <Sha>caeaf2a3fb3f636805fdd4881df4f9a539fff8f6</Sha>
```
```suggestion
      <Sha>802aa459394baa2451ffacc02219ba54ef77ca44</Sha>
```

hoyosjs reviewed (approved):
Minor suggestion given the rollback

ViktorHofer committed:
Update Version.Details.xml

ViktorHofer committed:
Suppress false positive package validation errors

ViktorHofer commented:
Reason for the package testing errors is https://github.com/dotnet/arcade/commit/582981cf4bc8de324d756b78c390c9610f20fb49#diff-4eed975465e70954faaaafdfff1b789ffb5346b1dbfde558f0d832e693eff73c which improves testing for RID specific assets. Had  to suppress false positive errors.

ViktorHofer commented:
@Anipik this dependency update PR needs to go in before we branch off. We should hold branching off until CI is working again and the PR is merged.

ViktorHofer commented:
CI is on the floor partially but is recovering now. The only change since the last run (which was mostly green) affects the AllConfigurations leg only which already passed. Merging to not require another build iteration.
